File: src/main/java/com/chessonline/Piece.java
package com.chessonline;

public abstract class Piece {
    protected String name;
    protected String color;
    protected String position;

    public Piece(String name, String color, String position) {
        this.name = name;
        this.color = color;
        this.position = position;
    }

    public abstract boolean move(String newPosition);

    public abstract boolean isMovementValid(String newPosition);

    public abstract boolean isCaptureMovementValid(String newPosition);

    public String getName() {
        return name;
    }

    public String getColor() {
        return color;
    }

    public String getPosition() {
        return position;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    protected static class Position {
        char col;
        int row;

        Position(String position) {
            this.col = position.charAt(0);
            this.row = position.charAt(1) - '0';
        }
    }
}

File: engine/pieces/piece.py
from abc import ABC, abstractmethod

class Piece(ABC):
    def __init__(self, name: str, color: str, position: str):
        self.name = name
        self.color = color
        self.position = position

    @abstractmethod
    def move(self, new_position: str) -> bool:
        pass

    @abstractmethod
    def is_movement_valid(self, new_position: str) -> bool:
        pass

    @abstractmethod
    def is_capture_movement_valid(self, new_position: str) -> bool:
        pass

    def get_name(self) -> str:
        return self.name

    def get_color(self) -> str:
        return self.color

    def get_position(self) -> str:
        return self.position

    def set_position(self, position: str):
        self.position = position

class Position:
    def __init__(self, position: str):
        self.col = position[0]
        self.row = int(position[1])